### PR TITLE
Edwin - Added user classes to weekly summary report

### PR DESCRIPTION
**Pull request by EdwinLauM**
# Description

Screen Shot 2023-07-25 at 12 04 08 PM

## Related PRS (if any):

This backend PR is related to the FE PR [PR #1042](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1042)

## Main changes explained:

- Made slight changes to backend for weekly summary request to now include the users role

## How to test:

1. check into current branch
2. do `npm run build` && `npm run start`

## Videos/Screenshots of changes:

Screen Shot 2023-07-25 at 12 08 47 PM

### CODE REVIEW

**Review by sxiong5 — APPROVED** [review]
Hey, I have tested the PR with [HighestGoodNetworkApp#1042](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1042) and left a comment. The feature worked fine.

**Review by BaileyM96 — APPROVED** [review]
Left a review on PR #1042, great work

**Review by winghojackyli — APPROVED** [review]
Hi @EdwinLauM , I left a review on #1042 and the PR works great.

**Review by veronicacheng2 — APPROVED** [review]
Tested together with #1042 and it works perfectly as stated.

**Comment by beblicarl** [comment]
This functionality works as intended

**Review by beblicarl — APPROVED** [review]

**Comment by Shreyj14** [comment]
Hi @EdwinLauM I tested this PR with frontend#1042, it worked as intended. Nice Job!

https://github.com/OneCommunityGlobal/HGNRest/assets/118795427/d4f6b915-2481-4437-9d09-34aca4b738d7

**Review by Shreyj14 — APPROVED** [review]

**Review by DavidC0126 — APPROVED** [review]
Hi,  I left a review on [PR#1042](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1042#pullrequestreview-1548747120) and the PR works perfectly.

**Comment by mounica-dingari-by** [comment]
Hi, the #PR1042 associated to this works as expected.

**Review by xaanders — APPROVED** [review]
I have tested it in frontend PR. Works good.

**Comment by iTvX** [comment]
I reviewed your code and test the pr function. It looks good. Great job!
CleanShot 2023-07-27 at 13 37 20@2x

**Review by vemanamadhu — APPROVED** [review]
Checked PR#1042 everything seems to be working fine. Good job

**Review by yaow62023 — APPROVED** [review]
I tested at PR# 1042, great work!

File: src/helpers/reporthelper.js [hunk 1]
```diff
@@ -80,6 +80,7 @@ const reporthelper = function () {
           weeklySummaryOption: 1,
           adminLinks: 1,
           bioPosted: 1,
+          role: 1,
```
Comment (iTvX) [inline src/helpers/reporthelper.js:83]: I always feel that the app is very laggy these days. Adding `role: 1` in very weekly report will also slow down the app. I suggest query this field separately when required.

**Review by mahamakhan — APPROVED** [review]
Looks good and works as moentioned.
image

**Review by zuhang2000 — APPROVED** [review]
I left a review on PR #1042. Everything work as expected.

**Review by shiva2096 — APPROVED** [review]
HI @EdwinLauM 

Test this with frontend PR. Functionality works as expected.

**Pull request merged**